// password-material-ui/lib/PasswordMaterialUi.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    }
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var lodash_1 = require("lodash");
var PasswordField_1 = require("./PasswordField");
var RemoveValue_1 = require("./RemoveValue");
var SeeHidePassword_1 = require("./SeeHidePassword");
var PasswordMaterialUi = (function (_super) {
    __extends(PasswordMaterialUi, _super);
    function PasswordMaterialUi() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.state = {
            isPasswordVisible: false
        };
        _this.handleChange = function (event) {
            return _this.props.onChange(event.target.value);
        };
        _this.handleRemoveValue = function () { return _this.props.onChange(""); };
        _this.handlePasswordVisibility = function (isPasswordVisible) {
            return _this.setState({
                isPasswordVisible: isPasswordVisible
            });
        };
        return _this;
    }
    PasswordMaterialUi.prototype.render = function () {
        var _a = this.props, id = _a.id, label = _a.label, placeholder = _a.placeholder, value = _a.value;
        return (React.createElement(PasswordField_1.default, { id: id, type: this.getType(), label: label, placeholder: placeholder, value: value, onChange: this.handleChange, startAdornment: this.getStartAdornament(), endAdornment: this.getEndAdornament() }));
    };
    PasswordMaterialUi.prototype.getType = function () {
        var type = this.props.type;
        if (lodash_1.isNil(type)) {
            return "text";
        }
        if (type !== "password") {
            return type;
        }
        return this.state.isPasswordVisible ? "text" : "password";
    };
    PasswordMaterialUi.prototype.getStartAdornament = function () {
        if (this.props.type !== "password") {
            return null;
        }
        return (React.createElement(SeeHidePassword_1.default, { isPasswordVisible: this.state.isPasswordVisible, onClick: this.handlePasswordVisibility }));
    };
    PasswordMaterialUi.prototype.getEndAdornament = function () {
        if (lodash_1.isEmpty(this.props.value)) {
            return null;
        }
        return React.createElement(RemoveValue_1.default, { onClick: this.handleRemoveValue });
    };
    return PasswordMaterialUi;
}(React.Component));
exports.default = PasswordMaterialUi;
