// password-material-ui/lib/SeeHidePassword.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    }
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var FontAwesome = require("react-fontawesome");
var InputAdornment_1 = require("@material-ui/core/InputAdornment/InputAdornment");
var style = {
    color: "#0080cc",
    cursor: "pointer"
};
var SeeHidePassword = (function (_super) {
    __extends(SeeHidePassword, _super);
    function SeeHidePassword() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.handleClick = function () { return _this.props.onClick(!_this.props.isPasswordVisible); };
        return _this;
    }
    SeeHidePassword.prototype.render = function () {
        return (React.createElement(InputAdornment_1.default, { style: style, position: "start" },
            React.createElement("div", { "aria-label": this.getAriaLabel(), onClick: this.handleClick },
                React.createElement(FontAwesome, { name: this.getIcon() }))));
    };
    SeeHidePassword.prototype.getAriaLabel = function () {
        var isPasswordVisible = this.props.isPasswordVisible;
        return isPasswordVisible ? "Hide the password" : "See the password";
    };
    SeeHidePassword.prototype.getIcon = function () {
        return this.props.isPasswordVisible ? "eye-slash" : "eye";
    };
    return SeeHidePassword;
}(React.Component));
exports.default = SeeHidePassword;
